batch/batch/aioclient.py

```diff
@@ -228,10 +235,11 @@ def __init__(self, session, url=None, token_file=None, token=None, headers=None)
                       attributes=b.get('attributes'))
                 for b in batches]
 
-    async def get_job(self, id):
-        j = await self._get('/jobs/{}'.format(id))
-        return Job(self,
-                   j['id'],
+    async def get_job(self, batch_id, job_id):
+        b = await self.get_batch(batch_id)
+        j = await self._get('/batches/{}/jobs/{}'.format(batch_id, job_id))
```

Comment:
Race condition potentially

Reply:
I think this is ok. There is no state stored in `self.get_batch()`. It's just the id and the attributes. The attributes are static. The batch is then double checked to still exist in the call we care about to get the Job. The options to fixing the double network call are caching the batch object in the client code or changing the job `to_dict` in the server to include batch information. I don't like the latter because it changes what I'd think should be the right interface for `Job.to_dict`. I can see where caching the batches on the client might be a good idea to avoid an extra network call. But that seems like overkill for where we're at right now.